File: 0169-majority-element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int c=0;
        int count=0;

        for(int n:nums){
            if(count==0){
                c=n;
            }
            if(n==c){
                count++;
            }else{
                count--;
            }
        }
        return c;
    }
}

File: 0169-majority-element/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        c=None
        count=0

        for i in nums:
            if count==0:
                c=i
            
            if c==i:
                count+=1
            else:
                count-=1
        return c
